src/ckb-daemon/usb_linux.c

```diff
@@ -319,6 +329,10 @@ void* os_inputmain(void* context){
         ioctl(fd, USBDEVFS_SUBMITURB, urbs + i);
     }
 
+    /// Allocate memory for the os_usbrecv() buffer and create the mutex
+    kb->interruptbuf = malloc(MSG_SIZE * sizeof(uchar));
+    pthread_mutex_init(&kb->interruptmutex, NULL);
```

Comment:
Missing error checking and handling, missing `pthread_mutex_destroy`.

Reply:
Thought the latter was added, will correct.
Although short of displaying a message, not sure what kind of error handling can be done in this case.

Reply:
Since the attributes on init() are NULL, instead of init() and destroy() consider using PTHREAD_MUTEX_INITIALIZER.